src/Modules/OrchardCore.Commerce/Views/OrderPart_Edit.cshtml

```diff
@@ -1,63 +1,122 @@
+@using System.Text.Json;
+@using OrchardCore.Commerce.Abstractions;
+@using OrchardCore.Commerce.Indexes;
+@using OrchardCore.Commerce.Models;
+@using OrchardCore.Commerce.MoneyDataType.Abstractions;
+@using OrchardCore.Commerce.Settings;
+@using OrchardCore.ContentManagement;
+@using YesSql;
+
 @model OrderPartViewModel
 
-<section id="shopping-cart">
-    <div class="panel-heading">
-        <label>@T["Ordered Items"]</label>
-    </div>
+@inject IMoneyService MoneyService
+@inject ISession Session
+@inject IPredefinedValuesProductAttributeService AttributesService
+
+@{
+    var availableCurrencyCodes = MoneyService.Currencies.Select(currency => currency.CurrencyIsoCode);
+    var defaultCurrencyCode = MoneyService.DefaultCurrency.CurrencyIsoCode ?? "USD";
 
-    @if (Model.LineItems?.Any() == true)
+    // Get all predefined attributes in the system by filtering for Products that have PriceVariantsPart attached.
+    var allProducts = await Session.Query<ContentItem, ProductPartIndex>().ListAsync();
+    var allPriceVariantProducts = allProducts.Where(product => product.As<PriceVariantsPart>() != null);
+
+    var availableAttributes = new Dictionary<string, List<string>>();
+    foreach (var priceVariant in allPriceVariantProducts)
     {
-        <div class="container">
-            <div class="d-none d-sm-flex row border-bottom mt-3 pb-2 font-weight-bold">
-                <div class="col-xs-12 col-sm-2">@T["Quantity"]</div>
-                <div class="col-xs-12 col-sm-4">@T["Product"]</div>
-                <div class="col-xs-12 col-sm-4">@T["Attributes"]</div>
-                <div class="col-xs-12 col-sm-1">@T["Price"]</div>
-            </div>
+        var attributes = AttributesService.GetProductAttributesRestrictedToPredefinedValues(priceVariant);
```

Comment:
One issue with this is that it only retrieves attributes when there already is a Price Variant Product among the Order's line items. Makes no sense to me as the query above specifically gets all Price Variant Products in the system regardless of what the Order contains.

This means that when the Order does not contain any Price Variant Products, attribute dropdowns are not available until a Price Variant Product is added. While this is sub-optimal, it also doesn't feel like a deal-breaker to me in terms of functionality.

Reply:
It's not clear to me why you need `allPriceVariantProducts`. I've added a text product attribute field with predefined values to Product, removed the `.Where(product => product.As<PriceVariantsPart>() != null)` from above, similarly use `productPart.ContentItem` in `HandleSelectedAttributes` and it seems to work:
![image](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/4007293/e71a9bbe-a817-49a2-bb48-4966326c9acb)

Reply:
I was under the impression only Price Variant Products need attributes as their prices are based on those. Is that not the purpose of attributes? Should any Product be able to have attributes?